### PR TITLE
Scheduled cron GH-action workflow for updating Boost version.

### PR DESCRIPTION
**Pull request by Yashwants19**
In this PR I have tried to automate the process of updating the boost version in our cmake script. As we are updating catch and cli11 using GH actions, this workflow will also open a PR for boost version update. For e.g. PR [#15](https://github.com/Yashwants19/mlpack/pull/15).

### CODE REVIEW

File: .github/workflows/update-boost-version.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: Update Boost Version
+on:
+  workflow_dispatch:
+  schedule:
+    - cron:  '0 10 * * *'
```
Comment (Yashwants19) [inline .github/workflows/update-boost-version.yaml:5]: This workflow will run every day at 10:00 UTC.

**Review by zoq — APPROVED** [review]
Looks great to me, there is also a new version, so once merged we should see an automated PR (10 UTC).

**Comment by zoq** [comment]
Nicely done, I guess we will see a new PR tomorrow.

**Comment by Yashwants19** [comment]
Thanks @zoq for the quick review and merge.

**Pull request merged**